### DISCUSSION TITLE
Copilot being able to read code below the current line

### DISCUSSION THREAD

**Discussion by asportnoy** [community/community#32465] [Copilot Conversations]
Earlier today, I was adding JSDoc type definitions to a bunch of existing functions:
```js
function doSomethingHere() {
  // ...
}

function doAnotherThingHere() {
  // ...
}
```

I added a comment above each one: `/** @type import('./index')["doSomethingHere"] */` (Replace `doSomethingHere` with the name of the function below it). There were a bunch of them, so I was expecting that after doing it for the first few, Copilot would get the idea and infer the name to put there from the function below it. However, while Copilot suggested the comment in the correct format, it was just trying to guess a name based on what it thinks would come next. It seemed like it was not aware of the code below where it was running.

I've also experienced a similar thing where I'm adding a function above some similar functions. If I were to add it below, Copilot would be able to complete the function correctly based on the ones above it. However, if I were to try to trigger that while above the "example" functions, Copilot has no idea about what I'm actually trying to do.

In summary, Copilot should be able to use code below where I'm currently using it in order to find patterns and continue them.
